pkg/builder/backup_repository_builer.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package builder
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	velerov1api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+)
+
+// BackupRepositoryBuilder builds BackupRepository objects.
+type BackupRepositoryBuilder struct {
+	object *velerov1api.BackupRepository
+}
+
+// ForBackupRepository is the constructor for a BackupRepositoryBuilder.
+func ForBackupRepository(ns, name string) *BackupRepositoryBuilder {
```

Comment:
Looks like this builder is used by the testing code only, while other builders in pkg/builder are used formally.
I compared the code in pkg/repository/ensurer.go, where the backupRepository object is created, the code there is like:
```
	repo := &velerov1api.BackupRepository{
		ObjectMeta: metav1.ObjectMeta{
			Namespace:    namespace,
			GenerateName: fmt.Sprintf("%s-%s-", volumeNamespace, backupLocation),
			Labels:       repoLabels(volumeNamespace, backupLocation, repositoryType),
		},
		Spec: velerov1api.BackupRepositorySpec{
			VolumeNamespace:       volumeNamespace,
			BackupStorageLocation: backupLocation,
		},
	}
```
We see differences, so I am afraid the code here doesn't work if we use it formally. Therefore:

- either we move it to the testing code
- or we make it the same with the code in ensurer.go and it's better to remove the code redundancy as well

Reply:
I defined one function in testing code